navigation/libandroid-navigation-ui/src/main/java/com/mapbox/services/android/navigation/ui/v5/NavigationView.java

```diff
@@ -0,0 +1,260 @@
+import com.mapbox.services.android.navigation.v5.offroute.OffRouteListener;
+import com.mapbox.services.android.navigation.v5.routeprogress.ProgressChangeListener;
+import com.mapbox.services.android.navigation.v5.routeprogress.RouteProgress;
+import com.mapbox.services.android.telemetry.location.LocationEngine;
+import com.mapbox.services.android.telemetry.location.LocationEngineListener;
+import com.mapbox.services.api.directions.v5.models.DirectionsResponse;
+import com.mapbox.services.api.directions.v5.models.DirectionsRoute;
+import com.mapbox.services.commons.models.Position;
+
+import retrofit2.Call;
+import retrofit2.Callback;
+import retrofit2.Response;
+
+public class NavigationView extends AppCompatActivity implements OnMapReadyCallback, LocationEngineListener,
+  ProgressChangeListener, OffRouteListener, Callback<DirectionsResponse> {
+
+  private MapView mapView;
+  private InstructionView instructionView;
+
+  private MapboxMap map;
+  private MapboxNavigation navigation;
+  private NavigationMapRoute mapRoute;
+  private NavigationCamera camera;
+  private LocationEngine locationEngine;
+  private LocationLayerPlugin locationLayer;
+
+  private Location location;
+  private Position destination;
+  private boolean unpackBundle;
+
+  @Override
+  protected void onCreate(@Nullable Bundle savedInstanceState) {
+    super.onCreate(savedInstanceState);
+    setContentView(R.layout.navigation_view_layout);
+    unpackBundle = savedInstanceState == null;
+    bind();
+
+    initMap(savedInstanceState);
+    initNavigation();
+  }
+
+  @Override
+  protected void onStart() {
+    super.onStart();
+    mapView.onStart();
+    if (locationLayer != null) {
+      locationLayer.onStop();
+    }
+  }
+
+  @Override
+  public void onResume() {
+    super.onResume();
+    mapView.onResume();
+  }
+
+  @Override
+  public void onPause() {
+    super.onPause();
+    mapView.onPause();
+  }
+
+  @Override
+  public void onLowMemory() {
+    super.onLowMemory();
+    mapView.onLowMemory();
+  }
+
+  @Override
+  protected void onStop() {
+    super.onStop();
+    mapView.onStop();
+    if (locationLayer != null) {
+      locationLayer.onStop();
+    }
+  }
+
+
+  @Override
+  protected void onDestroy() {
+    super.onDestroy();
+    mapView.onDestroy();
+    if (navigation != null) {
+      navigation.onDestroy();
+    }
+  }
+
+  @Override
+  protected void onSaveInstanceState(Bundle outState) {
+    super.onSaveInstanceState(outState);
+    mapView.onSaveInstanceState(outState);
+  }
+
+  @Override
+  public void onMapReady(MapboxMap mapboxMap) {
+    map = mapboxMap;
+    initLocation();
+    initLocationLayer();
+    initRoute();
+    initCamera();
+  }
+
+  @SuppressWarnings({"MissingPermission"})
+  @Override
+  public void onConnected() {
+    locationEngine.requestLocationUpdates();
+  }
+
+  @Override
+  public void onLocationChanged(Location location) {
+    this.location = location;
+    if (unpackBundle) {
```

Comment:
Does this really need to be checked every location update or could we place this inside onCreate or somewhere else?